aikiblog/views.py

```diff
@@ -79,14 +78,52 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class TrainingDetail(generic.DetailView):
-    model = Training
+class TrainingDetail(generic.FormView):
+    form_class = TrainingCommentForm
+    template_name = 'aikiblog/training_detail.html'
 
-    def get_context_data(self, **kwargs):
-        context = super(TrainingDetail, self).get_context_data(**kwargs)
-        context['all_trainings'] = Training.objects.order_by('-date')[:7]
+    def get_success_url(self):
+        slug = self.kwargs['slug']
+        url = reverse('training-detail', kwargs={'slug': slug})
+        return url
 
-        return context
+    def get(self, request, *args, **kwargs):
+        form = self.get_form()
+        return self.render_to_response(
+            self.get_context_data(form=form))
+
+    def post(self, request, *args, **kwargs):
+        form = TrainingCommentForm(request.POST or None)
+        if form.is_valid():
+            user = request.user
+            training_comment = TrainingComment.objects.create(
+                training_id=self.get_training(),
+                posted_date=datetime.datetime.now(),
+                text=form.cleaned_data['text'],
+                author=user
+            )
```

Comment:
A tego nigdzie nie używasz. To poprawka dla Ciebie, bo nie wiem, co chciałbyś z tym zrobić.